src/hooks/useLedgerImport.ts

```diff
@@ -0,0 +1,227 @@
+            complete: () => {},
+            error: error => {
+              logger.debug('error pairing ', { error }, DebugContext.ledger);
+              setPairingStatus(error);
+            },
+            next: async e => {
+              if (e.type === 'add') {
+                const device = e.descriptor;
+                // prevent duplicate alerts
+                if (currentDeviceId === device.id) {
+                  return null;
+                }
+                // set the current device id to prevent duplicate alerts
+                setPairingStatus(LEDGER_IMPORT_STATUS.FOUND);
+                currentDeviceId = device.id;
+
+                /*
+                Alert.alert(
+                  'device found',
+                  `do u wanna connect to ${device.name} - ${device.id}`,
+                  [
+                    {
+                      onPress: () => {
+                        // if a user cancels this how do we want to handle?
+                        return null;
+                      },
+                      style: 'cancel',
+                      text: 'no',
+                    },
+                    {
+                      onPress: async () => {
+                        setPairingStatus(LEDGER_IMPORT_STATUS.PAIRING);
+                        try {
+                          // should i retry at this point?
+                          await openLedgerTransport(device.id);
+                          handlePairSuccess(device.id);
+                        } catch (e) {
+                          logger.debug(
+                            'error pairing ',
+                            {},
+                            DebugContext.ledger
+                          );
+                          handlePairError(e as Error);
+                        }
+                      },
+                      text: 'yes/pair',
+                    },
+                  ]
+                );
+                */
+                setPairingStatus(LEDGER_IMPORT_STATUS.PAIRING);
+                try {
+                  const transport = await TransportBLE.open(device.id);
+                  handlePairSuccess(device.id);
+                } catch (e) {
+                  logger.debug('error pairing ', {}, DebugContext.ledger);
+                  handlePairError(e as Error);
+                }
+              } else {
+                Alert.alert('error connecting', JSON.stringify(e, null, 2));
+              }
+            },
+          });
+          setListenerSubscription(newListener);
+        }
+      },
+    });
+    setObserverSubscription(newObserver);
+  }, [handlePairError, handlePairSuccess]);
+
+  /**
+   * Init ledger device search
+   * Reset conn for testing purposes when sheet is closed
+   */
+
+  useEffect(() => {
+    const asyncFn = async () => {
+      logger.debug(
+        '[LedgerImport] - init device polling',
+        {},
+        DebugContext.ledger
+      );
+      let isBluetothEnabled = true;
```

Comment:
maybe would be cleaner as `const isBluetoothEnabled = IS_ANDROID ? (await checkAndRequestAndroidBluetooth()) : true`